kale/loaddata/multi_domain.py

```diff
@@ -229,3 +232,249 @@ def _split_dataset_few_shot(dataset, n_fewshot, random_state=None):
+        self.split_ratio = split_ratio
+        if split_train_test:
+            self.train_idx, self.test_idx = split_by_domains(self.domain_labels, 2, [split_ratio])
+        else:
+            self.train_idx = None
+            self.test_idx = None
+
+    @staticmethod
+    def _find_classes(directory: str) -> Tuple[List[str], Dict[str, int]]:
+        """
+            Finds the class folders in a dataset.
+            Args:
+                directory (string): Directory path.
+            Returns:
+                tuple: (classes, class_to_idx) where classes are relative to (dir), and class_to_idx is a dictionary.
+            Ensures:
+                No class is a subdirectory of another.
+            """
+        classes = [d.name for d in os.scandir(directory) if d.is_dir()]
+        classes.sort()
+        class_to_idx = {cls_name: i for i, cls_name in enumerate(classes)}
+        return classes, class_to_idx
+
+    def __getitem__(self, index: int) -> Tuple:
+        """
+            Args:
+                index (int): Index
+            Returns:
+                tuple: (sample, target, domain) where target is class_index of the target class.
+            """
+        path, target, domain = self.samples[index]
+        sample = self.loader(path)
+        if self.transform is not None:
+            sample = self.transform(sample)
+        if self.target_transform is not None:
+            target = self.target_transform(target)
+        if self.return_domain_label:
+            return sample, target, domain
+        else:
+            return sample, target
+
+    def __len__(self) -> int:
+        return len(self.samples)
+
+    def get_train(self):
+        if self.split_train_test:
+            return torch.utils.data.Subset(self, self.train_idx)
+        else:
+            raise ValueError("The dataset has no train split yet.")
+
+    def get_test(self):
+        if self.split_train_test:
+            return torch.utils.data.Subset(self, self.test_idx)
+        else:
+            raise ValueError("The dataset has no test split yet.")
+
+
+def make_multi_domain_set(
+    directory: str,
+    class_to_idx: Dict[str, int],
+    domain_to_idx: Dict[str, int],
+    extensions: Optional[Tuple[str, ...]] = None,
+    is_valid_file: Optional[Callable[[str], bool]] = None,
+) -> List[Tuple[str, int, int]]:
+    """Generates a list of samples of a form (path_to_sample, class, domain).
+    Args:
+        directory (str): root dataset directory
+        class_to_idx (Dict[str, int]): dictionary mapping class name to class index
+        domain_to_idx (Dict[str, int]): dictionary mapping d name to class index
+        extensions (optional): A list of allowed extensions.
+            Either extensions or is_valid_file should be passed. Defaults to None.
+        is_valid_file (optional): A function that takes path of a file
+            and checks if the file is a valid file
+            (used to check of corrupt files) both extensions and
```

Comment:
See the above similar comments.

Also why break lines so frequently, it is far less than 120 chars.